client/map_client.go

```diff
@@ -52,6 +52,16 @@ func (c *MapClient) GetAndVerifyLatestMapRoot(ctx context.Context) (*types.MapRo
 	return c.VerifySignedMapRoot(rootResp.GetMapRoot())
 }
 
+// GetAndVerifyMapRootByRevision verifies and returns the map root with the given revision.
+func (c *MapClient) GetAndVerifyMapRootByRevision(ctx context.Context, revision int64) (*types.MapRootV1, error) {
+	rootResp, err := c.Conn.GetSignedMapRootByRevision(ctx, &trillian.GetSignedMapRootByRevisionRequest{MapId: c.MapID, Revision: revision})
+	if err != nil {
+		s := status.Convert(err)
+		return nil, status.Errorf(s.Code(), "GetSignedMapRootByRevision(%v): %v", c.MapID, s.Message())
+	}
+	return c.VerifySignedMapRoot(rootResp.GetMapRoot())
```

Comment:
Might want to include a check that the returned revision was indeed the one that we asked for.

Reply:
Done.